documentation/projects.md

```diff
@@ -271,10 +273,186 @@ BOLT_PROJECT='~/project/my_project' bolt command run uptime -t target1
 ```
 
 > **Note:** The `BOLT_PROJECT` environment variable takes precedence over the
-> `--configfile` CLI option. 
+> `--configfile` CLI option.
+
+## Migrate a Bolt project
+
+After upgrading to a newer version of Bolt, you might find that a Bolt project
+you created in the past no longer works as expected. Or there might be new
+features that you want to use in an existing project. The `migrate` command
+allows you to update old Bolt projects so that you can use them with the latest
+Bolt release.
+
+Currently, the `migrate` command:
+- Updates inventory files from Bolt version 1.
+- Updates projects to use `bolt-project.yaml` and `inventory.yaml` instead of `bolt.yaml`.
+- Updates projects to implement module dependency management. Dependency
+  management was introduced in Bolt 2.30.0.
+
+To migrate a project:
+
+_\*nix shell command_
+
+```shell
+bolt project migrate
+```
+
+_PowerShell cmdlet_
+
+```powershell
+Update-BoltProject
+```
+
+The migrate command modifies files in your project and does not preserve
+comments or formatting. Before using the command, **make sure to use source
+control or backup your projects**.
+
+### How Bolt updates inventory files
+
+Bolt locates the inventory file for the current Bolt project and migrates it
+in place.
+
+The updates change the following keys in your inventory file:
+- `nodes` becomes `targets`.
+- The `name` key in a `Target` object becomes a `uri` key.
+
+For example, the following inventory file from Bolt version 1:
+
+```yaml
+groups:
+  - name: linux
+    nodes:
+      - name: target1.example.com
+        alias: target1
+      - name: target2.example.com
+        alias: target2
+```
+
+Becomes:
+
+```yaml
+groups:
+  - name: linux
+    targets:
+      - uri: target1.example.com
+        alias: target1
+      - uri: target2.example.com
+        alias: target2
+```
+
+### How Bolt updates project configuration
+
+Bolt locates a `bolt.yaml` file and moves its configuration to a new `bolt-project.yaml` file
+and the `inventory.yaml` file. It then deletes the `bolt.yaml` file.
+
+Project-specific configuration is moved to the `bolt-project.yaml` file, while transport
+configuration is moved to the top-level `config` key in the `inventory.yaml` file. If the
+`inventory.yaml` file has an existing top-level `config` key, the transport configuration
+from `bolt.yaml` is deep merged, with the configuration in `inventory.yaml` having higher
+precedence.
+
+For example, a project with the following `bolt.yaml` and `inventory.yaml` files:
+
+'''yaml
```

Comment:
Euugh. How did that happen? A bunch of these are single quotes instead of backticks. Fixed!